pyproject.toml

```diff
@@ -31,7 +31,7 @@ classifiers=[
     "Programming Language :: Python :: 3.12",
     "Topic :: Scientific/Engineering",
 ]
-dependencies = ["h5py", "packaging"]
+dependencies = ["h5py", "pyfive", "packaging"]
```

Comment:
This is worth discussing!

I think pyfive should be an optional dependency for h5netcdf (added to the "test" dependencies below) -- and if pyfive works as a drop-in replacement, probably we should also make h5py, optional, too!

This would be a significant breaking change for h5netcdf users (they will have to add another underlying HDF5 library to their dependencies), but I think it's the right long-term choice. We should note it prominently in the change log.

Also, is there a minimum pyfive version that will work? If so, we should declare it, e.g., `"pyfive>=0.5"`

Reply:
I agree with all of this. I'll add some code that only tries to import pyfive if `backend="pyfive"`, and fails gracefully if the package cannot be found.

The current minimum version is indeed `0.5.0`.

Reply:
Ha! That code is already there :). All I need to do is remove "pyfive" from the dependencies list and put it in project.optional-dependencies

Reply:
Done: 44338f6bc972a7b806716ceeb20d7e32207c4466

Reply:
For my tuppence worth, I think the minimum pyfive should be 0.5.0 as it's the first with lazy loading etc ...